system-test/managed_writer_client_test.ts

```diff
@@ -857,6 +865,266 @@ describe('managedwriter.WriterClient', () => {
+          const [table] = await bigquery
+            .dataset(flakyDatasetId)
+            .createTable(flakyTableId, {
+              schema,
+              location: flakyRegion,
+            });
+          projectId = table.metadata.tableReference.projectId;
+          parent = `projects/${projectId}/datasets/${flakyDatasetId}/tables/${flakyTableId}`;
+
+          const connection = await client.createStreamConnection({
+            streamType: managedwriter.PendingStream,
+            destinationTable: parent,
+          });
+
+          const writer = new JSONWriter({
+            connection,
+            protoDescriptor,
+          });
+
+          const iterations = new Array(50).fill(1);
+          let offset = 0;
+          for (const _ of iterations) {
+            const rows = generateRows(10);
+            const pw = writer.appendRows(rows, offset);
+            try {
+              const res = await pw.getResult();
+              assert.equal(res.error, null);
+            } catch (err) {
+              console.error('found error trying to send rows', err);
+              throw err;
+            }
+            offset += 10;
+            connection.close(); // Close connection on every append to trigger reconnection
+          }
+
+          const res = await connection.finalize();
+          connection.close();
+          assert.equal(res?.rowCount, 500);
+
+          writer.close();
+        } finally {
+          client.close();
+        }
+      }).timeout(2 * 60 * 1000);
+    });
+
+    describe('should manage to send data in parallel', () => {
+      it('every 10 request drops the connection', async () => {
+        bqWriteClient.initialize();
+        const client = new WriterClient();
+        client.enableWriteRetries(true);
+        client.setMaxRetryAttempts(10);
+        client.setClient(bqWriteClient);
+
+        try {
+          const flakyTableId = generateUuid() + '_reconnect_on_close';
+          const [table] = await bigquery
+            .dataset(flakyDatasetId)
+            .createTable(flakyTableId, {
+              schema,
+              location: flakyRegion,
+            });
+          projectId = table.metadata.tableReference.projectId;
+          parent = `projects/${projectId}/datasets/${flakyDatasetId}/tables/${flakyTableId}`;
+
+          const connection = await client.createStreamConnection({
+            streamType: managedwriter.PendingStream,
+            destinationTable: parent,
+          });
+
+          const writer = new JSONWriter({
+            connection,
+            protoDescriptor,
+          });
+
+          const pendingWrites: PendingWrite[] = [];
+          const iterations = new Array(50).fill(1);
+          let offset = 10;
+          for (const _ of iterations) {
+            const rows = generateRows(10);
+            const pw = writer.appendRows(rows);
+            pendingWrites.push(pw);
+            offset += 10;
```

Comment:
lint seems correct here, it's not clear if you intended to use this for an assertion of if you mean to do offset-based appends.

Reply:
good catch, I had to do some changes to be able to make it work properly on an async scenario.